src/js/components/AutolinkComponent.jsx

```diff
@@ -0,0 +1,67 @@
+var React = require("react/addons");
+
+/**
+ * Credits: https://github.com/banyan/react-autolink
+ * Lifting off some code as the package is not compatible with React 0.13
+ * TODO remove and add to dependecies
+ * after https://github.com/mesosphere/marathon/issues/2710
+ */
+
+const URLDelimiter = /(https?:\/\/[^\s]+)/gi;
+
+const startsWith = (str, prefix) => {
+  return str.slice(0, prefix.length) === prefix;
+};
+
+var AutolinkComponent = React.createClass({
+  displayName: "AutolinkComponent",
+
+  propTypes: {
+    options: React.PropTypes.object,
+    text: React.PropTypes.string.isRequired
+  },
+
+  getDefaultProps: function () {
+    return {
+      options: {
+        target: "_blank"
+      }
+    };
+  },
+
+  autolink: function () {
+    return this.props.text
+      .split(URLDelimiter)
+      .map(word => {
+        var match = word.match(URLDelimiter);
+        if (match) {
+          let url = match[0];
+          let segments = url.split("/");
+          // no scheme given, so check host portion length
+          if (segments[1] !== "" && segments[0].length < 5) {
+            return word;
+          }
+
+          return React.createElement("a",
+            Object.assign({}, {
+              key: `autolink-${url}`,
+              href: startsWith(url, "http")
```

Comment:
Instead of startsWith you could use a much more performant RegExp: 
```JavaScript
href: /^http/i.test(url)
```
It's a very simple and readable Regex, so I see no problem with lack of descriptiveness.